speechbrain/lobes/models/beats.py

```diff
@@ -0,0 +1,2024 @@
+            0.0  # probability of dropping a tarnsformer layer
+        )
+        self.dropout_input: float = (
+            0.0  # dropout to apply to the input (after feat extr)
+        )
+
+        # positional embeddings
+        self.conv_pos: int = (
+            128  # number of filters for convolutional positional embeddings
+        )
+        self.conv_pos_groups: int = (
+            16  # number of groups for convolutional positional embedding
+        )
+
+        # relative position embedding
+        self.relative_position_embedding: bool = (
+            False  # apply relative position embedding
+        )
+        self.num_buckets: int = (
+            320  # number of buckets for relative position embedding
+        )
+        self.max_distance: int = (
+            1280  # maximum distance for relative position embedding
+        )
+        self.gru_rel_pos: bool = (
+            False  # apply gated relative position embedding
+        )
+
+        # label predictor
+        self.finetuned_model: bool = (
+            False  # whether the model is a fine-tuned model.
+        )
+        self.predictor_dropout: float = (
+            0.1  # dropout probability for the predictor
+        )
+        self.predictor_class: int = 527  # target class number for the predictor
+
+        if cfg is not None:
+            self.update(cfg)
+
+    def update(self, cfg: dict):
+        """
+        Updates the instance's attributes with key-value pairs from a given configuration dictionary.
+
+        Arguments
+        ---------
+        cfg : dict
+            A dictionary containing the configuration values to update the instance with.
+        """
+        self.__dict__.update(cfg)
+
+
+class BEATs(nn.Module):
+    """
+    BEATs: Bidirectional Encoder Representations from Audio Transformers.
+
+    This class implements the BEATs model, which processes audio signals for feature extraction
+    or downstream tasks. The model supports loading from a checkpoint, applying normalization,
+    and optionally freezing parameters.
+
+    Arguments
+    ---------
+    ckp_path : str, optional
+        Path to the checkpoint file. If None, the model initializes without pre-trained weights.
+        You could download the checkpoints from : https://github.com/microsoft/unilm/tree/master/beats
+    freeze : bool, optional (default: False)
+        If True, the model parameters are frozen and the model is set to evaluation mode.
+    output_all_hiddens : bool, optional (default: False)
+        If True, the forward function outputs hidden states from all transformer layers.
+        For example BEATs_iter3 has 12 transformer layers and the output is of shape (13, B, T, C),
+        where a projection of the CNN output is added to the beginning.
+        If False, the forward function outputs the hidden states only from the last transformer layer.
+
+    Example
+    -------
+    >>> audio = torch.randn(4, 10000)  # Batch of 4 audio signals
```

Comment:
```
audio = torch.randn(4, 10000)  # Batch of 4 audio signals
length = torch.tensor([1.0, 0.5, 0.75, 1.0])
model = BEATs("BEATs_iter1_finetuned_on_AS2M_cpt2.pt")
outputs = model.extract_features(audio, length)
print(outputs.shape)
>>> AttributeError: 'tuple' object has no attribute 'shape'
```

When loading a pretrained model, I guess you do not need the predictor but just the embedding model (see line 2005).

Reply:
It happens when the finetuned version is loaded, so it also returns the prob_log of the prediction. I changed the code, so it could work with both finetuned and self-supervised checkpoints